libsolidity/formal/SMTEncoder.cpp

```diff
@@ -196,7 +201,11 @@ void SMTEncoder::inlineModifierInvocation(ModifierInvocation const* _invocation,
 	if (auto modifier = dynamic_cast<ModifierDefinition const*>(_definition))
 	{
 		if (modifier->isImplemented())
+		{
+			pushInlineFrame(modifier);
 			modifier->body().accept(*this);
+			popInlineFrame(modifier);
```

Comment:
I like the `pushInlineFrame` and `popInlineFrame` pair and from what I can see it is used in the same places that my `pushPathCondition/popPathCondition` pair needed for BMC. I think we can unify this, moving the call to `push/popPathCondition` inside the `push/popInlineFrame`. (It would require small modification for this part here, as we also need to do the push/pop for the if-case below this one).